client/dashboard/me/active-subscriptions/active-subscription-expiry.tsx

```diff
@@ -0,0 +1,323 @@
+	// to retrieve the correct taxName. For now, we are using a fallback tax
+	// name with context, to prevent mis-translation.
+	// translators: Shortened form of 'Sales Tax', not a country-specific tax name
+	const taxName = __( 'tax' );
+
+	/* translators: %s is the name of taxes in the country (eg: "VAT" or "GST"). */
+	const excludeTaxStringAbbreviation = sprintf( __( '(excludes %s)' ), [ taxName ] );
+
+	/* translators: %s is the name of taxes in the country (eg: "VAT" or "GST"). */
+	const excludeTaxStringTitle = sprintf( __( 'Renewal price excludes any applicable %s' ), [
+		taxName,
+	] );
+
+	if ( purchase.partner_name ) {
+		// translators: partnerName is the name of the partner service who manages this product
+		return sprintf( __( 'Managed by %(partnerName)s' ), {
+			partnerName: purchase.partner_name,
+		} );
+	}
+
+	if (
+		isDisconnectedSite &&
+		isTemporarySitePurchase( purchase ) &&
+		purchase.product_type === 'jetpack'
+	) {
+		return (
+			<>
+				<span>{ __( 'Activate your product license key' ) }</span>
+				<br />
+				<ExternalLink href="https://jetpack.com/support/activate-a-jetpack-product-via-license-key/">
+					{ __( 'Learn more' ) }
+				</ExternalLink>
+			</>
+		);
+	}
+
+	const isA4APurchase = isA4ATemporarySitePurchase( purchase );
+	const temporarySitePurchaseProductTypes = [ 'saas_plugin', 'jetpack', 'akismet' ];
+	const isKnownTemporarySiteProductType =
+		isTemporarySitePurchase( purchase ) &&
+		temporarySitePurchaseProductTypes.includes( purchase.product_type );
+
+	if ( isDisconnectedSite && ! isA4APurchase && ! isKnownTemporarySiteProductType && isJetpack ) {
+		return <span>{ __( 'Disconnected from WordPress.com' ) }</span>;
+	}
+
+	if ( isDisconnectedSite && ! isA4APurchase && ! isKnownTemporarySiteProductType ) {
+		return (
+			<span>
+				{ createInterpolateElement(
+					__(
+						'You no longer have access to this site and its purchases. <button>Contact support</button>'
+					),
+					{
+						button: <a href="/help/contact" />,
+					}
+				) }
+			</span>
+		);
+	}
+
+	if ( purchase.is_iap_purchase && purchase.iap_purchase_management_link ) {
+		return createInterpolateElement(
+			__(
+				'This product is an in-app purchase. You can manage it from within <managePurchase>the app store</managePurchase>.'
+			),
+			{
+				managePurchase: <a href={ purchase.iap_purchase_management_link } />,
+			}
+		);
+	}
+
+	const isIntroductoryOfferFreeTrial = purchase.introductory_offer?.cost_per_interval === 0;
+	if (
+		purchase.introductory_offer?.is_within_period &&
+		isIntroductoryOfferFreeTrial &&
+		isRenewing( purchase )
+	) {
+		return createInterpolateElement(
+			sprintf(
+				// translators: date is a formatted date, amount is a currency amount, and excludeTaxStringAbbreviation is something like "excludes VAT"
+				__(
+					'Free trial ends on <span>%(date)s</span>, renews automatically at %(amount)s <abbr>%(excludeTaxStringAbbreviation)s</abbr>'
```

Comment:
Team review: Consider moving HTML tags inside components instead of the string. For instance: instead of `<abbr>%(excludeTaxStringAbbreviation)s</abbr>`, use `<excludeTaxStringAbbreviation />`.

Reply:
See https://github.com/Automattic/wp-calypso/pull/103700#discussion_r2108720909.